src/structs.jl

```diff
@@ -29,9 +29,16 @@ function _prepare_read(str::AbstractString, ::Type{T}) where {T}
     invalid(error, buf, pos, T)
 end
 
-function read(str::AbstractString, ::Type{T}; kw...) where {T}
+function read(str::AbstractString, ::Type{T}; jsonlines::Bool=false, kw...) where {T}
     buf, pos, len, b = _prepare_read(str, T)
-    pos, x = read(StructType(T), buf, pos, len, b, T; kw...)
+    if jsonlines
+        if StructType(T) != ArrayType()
+            throw(ArgumentError("expect StructType($T) == ArrayType() when jsonlines=True"))
```

Comment:
```suggestion
            throw(ArgumentError("expect StructType($T) == StructTypes.ArrayType() when jsonlines=true"))
```

Reply:
Hmmm, I wonder if the interface should be that you just pass the struct type `T` that will be found on each line, then we'll just pass `Vector{T}` to be parsed. It seems a little more convenient. Though I guess this way allows passing somethign other than `Vector{T}` if that's what you want. @mcmcgrath13, any thoughts on this?

Reply:
Maybe somewhere in between.  I like the flexibility of passing a custom array type, but if nothing is passed instead of `throw`, maybe log a warning and then wrap the `T` in `Vector{T}`.

Reply:
I have been thinking about this.

First, I don't think that logging a warning is a good idea.  If we want to allow users to just pass in `T`, then warning them not to seems annoying.

So I think the question comes down to whether or not we want to allow users to specify an array type other than a `Vector{T}`. My preference is to allow this, as I frequently use alternate array types and it would be annoying to load the data in and then transform it to another array type.

That would argue for the current syntax.

We could possibly allow either syntax, but that would preclude jsonlines files with arrays on each line, which are not supported by this PR, but which we actually have where I work and which I would  like to try to support in a future PR.

Thoughts?

Reply:
Given the array of arrays case, I agree throwing is probably the safest/most consistent bet.